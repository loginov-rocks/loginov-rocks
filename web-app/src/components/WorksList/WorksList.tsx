import { GitHubRepo } from '@loginov-rocks/loginov-rocks-shared';
import * as React from 'react';

import { WorksListItem } from 'components/WorksListItem';

interface Props {
  gitHubRepos?: GitHubRepo[];
  works: string[];
}

export const WorksList: React.FunctionComponent<Props> = ({ gitHubRepos, works }) => (
  <ul>
    {/* eslint-disable-next-line react/no-array-index-key */}
    {works.map((work, index) => <WorksListItem gitHubRepos={gitHubRepos} key={index} work={work} />)}
  </ul>
);

WorksList.defaultProps = {
  gitHubRepos: [],
};
