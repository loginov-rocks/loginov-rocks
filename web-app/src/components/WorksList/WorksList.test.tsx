/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import { WorksList } from './WorksList';

it('matches snapshot', () => {
  const wrapper = shallow(<WorksList works={['Some work']} />);

  expect(wrapper).toMatchSnapshot();
});
