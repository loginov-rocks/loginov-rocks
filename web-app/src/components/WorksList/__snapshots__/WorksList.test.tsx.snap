// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<ul>
  <WorksListItem
    gitHubRepos={Array []}
    key="0"
    work="Some work"
  />
</ul>
`;
